MastodonSDK/Sources/MastodonUI/View/Content/StatusCardControl.swift

```diff
@@ -0,0 +1,195 @@
+        }
+
+        titleLabel.text = card.title
+        linkLabel.text = card.url?.host
+        imageView.contentMode = .center
+
+        imageView.sd_setImage(
+            with: card.imageURL,
+            placeholderImage: icon(for: card.layout)
+        ) { [weak self] image, _, _, _ in
+            if image != nil {
+                self?.imageView.contentMode = .scaleAspectFill
+            }
+
+            self?.containerStackView.setNeedsLayout()
+            self?.containerStackView.layoutIfNeeded()
+        }
+
+        updateConstraints(for: card.layout)
+    }
+
+    public override func didMoveToWindow() {
+        super.didMoveToWindow()
+
+        if let window = window {
+            layer.borderWidth = 1 / window.screen.scale
+        }
+    }
+
+    private func updateConstraints(for layout: Layout) {
+        guard layout != self.layout else { return }
+        self.layout = layout
+
+        NSLayoutConstraint.deactivate(layoutConstraints)
+
+        switch layout {
+        case .large(let aspectRatio):
+            containerStackView.alignment = .fill
+            containerStackView.axis = .vertical
+            layoutConstraints = [
+                imageView.widthAnchor.constraint(
+                    equalTo: imageView.heightAnchor,
+                    multiplier: aspectRatio
+                )
+                // This priority is important or constraints break;
+                // it still renders the card correctly.
+                .priority(.defaultLow - 1),
+            ]
+        case .compact:
+            containerStackView.alignment = .center
+            containerStackView.axis = .horizontal
+            layoutConstraints = [
+                imageView.heightAnchor.constraint(equalTo: heightAnchor),
+                imageView.widthAnchor.constraint(equalToConstant: 85),
+                heightAnchor.constraint(equalToConstant: 85).priority(.defaultLow - 1),
+                heightAnchor.constraint(greaterThanOrEqualToConstant: 85)
+            ]
+        }
+
+        NSLayoutConstraint.activate(layoutConstraints)
+    }
+
+    private func icon(for layout: Layout) -> UIImage? {
+        switch layout {
+        case .compact:
+            return UIImage(systemName: "newspaper.fill")
+        case .large:
+            let configuration = UIImage.SymbolConfiguration(pointSize: 32)
+            return UIImage(systemName: "photo", withConfiguration: configuration)
+        }
+    }
+
+    private func apply(theme: Theme) {
+        layer.borderColor = theme.separator.cgColor
+        imageView.backgroundColor = theme.systemElevatedBackgroundColor
+    }
+}
+
+private extension StatusCardControl {
+    enum Layout: Equatable {
+        case compact
+        case large(aspectRatio: CGFloat)
+    }
+}
+
+private extension Card {
+    var layout: StatusCardControl.Layout {
+        return width == height || image == nil
+        ? .compact
+        : .large(aspectRatio: CGFloat(width) / CGFloat(height))
+    }
```

Comment:
I saw a forums.swift.org link with a 397×400 image. This change will treat slightly non-square images as square too.

```suggestion
    var layout: StatusCardControl.Layout {
        let aspectRatio = CGFloat(width) / CGFloat(height)
        return abs(aspectRatio - 1) < 0.05 || image == nil
        ? .compact
        : .large(aspectRatio: aspectRatio)
    }
```